app/vlinsert/loki/loki_json.go

```diff
@@ -140,7 +141,7 @@ func parseJSONRequest(data []byte, lmp insertutils.LogMessageProcessor) (int, er
 			if err != nil {
 				return rowsIngested, fmt.Errorf("unexpected contents of `values` item; want array; got %q", line)
 			}
-			if len(lineA) != 2 {
+			if len(lineA) < 2 {
```

Comment:
Could you please also attach metadata to the log fields? 

 As far as I see, it's already supported for protobuf parsing: 
https://github.com/VictoriaMetrics/VictoriaMetrics/blob/312ea226f3bcc0bc9cb7f8825a610f3ef2bbec94/app/vlinsert/loki/loki_protobuf.go#L108
```
			for _, lp := range e.StructuredMetadata {
				fields.fields = append(fields.fields, logstorage.Field{
					Name:  lp.Name,
					Value: lp.Value,
				})
			}
```

Reply:
Thanks! I appended it below. Please check and I'd like to have some feedback regarding the usage of fastcache. Not sure if it's correct but the test cases LGTM.